types/node.js

```diff
@@ -51,22 +50,19 @@ class Node extends Service {
    * @param {Object|String} settings Label for the trusted messages, or a configuration object.
    */
   trust (source, settings = {}) {
-    let extra = '';
-
-    if (typeof settings === 'string') {
-      extra = `[${settings}]`;
-    }
+    const self = this;
+    const extra = (typeof settings === 'string') ? `[${settings}] ` : '';
 
     source.on('debug', function (debug) {
-      console.debug(`[FABRIC:DEBUG] ${extra}`, debug);
+      self.emit('debug', `[FABRIC:DEBUG] ${extra}${debug}`);
     });
 
     source.on('info', function (info) {
       console.log(`[FABRIC:INFO] ${extra}`, info);
     });
 
     source.on('log', function (log) {
-      console.log(`[FABRIC:LOG] ${extra}`, log);
+      self.emit('log', `[FABRIC:LOG] ${extra}${log}`);
```

Comment:
Is there a difference between `info` and `log` levels? IMO they are the same, but named differently, and in a monitoring scenario, they would be routed identically. Perhaps we pick one and go with that? I'd vote `info`, with `log` being set deprecated.

Reply:
Alternately, we can have a `log` event with a handler signature like `function (logLevel, ...args) { ... }`.  These events can then be wired into the logger to produce log out in the desired format.

One could even say that the `logger` class would contain a handler that can be wired up to a class instance as follows `instance.on('log', logger.onLogEvent)`.